app/models/changeset/pull_request.rb

```diff
@@ -3,11 +3,14 @@ class Changeset::PullRequest
   RISKS_SECTION = /#+\s+Risks.*\n/i
 
   # Matches URLs to JIRA issues.
-  JIRA_ISSUE = %r[https://\w+\.atlassian\.net/browse/[\w-]+]
+  JIRA_ISSUE_URL = %r[https?:\/\/[\da-z\.\-]+\.[a-z\.]{2,6}\/browse\/[A-Z][A-Z\d]+-\d+]
```

Comment:
> This would also retain the flexibility of referencing issues from multiple JIRA instances by their full URLs, if that is a valid use case.

I guess this is what I'm really questioning. The main application of this I that I can see is for organisations that switch JIRA instances or move between on-premise and cloud. IMO this is an edge case, but I suppose since you've done the work it would be better to keep it.

Can you add a specific test case that make use of this functionality (i.e., pulling out issues from multiple JIRA domains?